### PR TITLE
Change default .erlang loading

### PR DESCRIPTION
**Pull request by dgud**
Loading "hidden" '.erlang' file from the current directory surprises users, scripts and tests.

In tools 'escript', 'erlc' and 'dialyzer' it was impossible to opt out on loading '.erlang' files,
and there is no need to load the file there, in escript the users can invoke c:erlangrc() and
options and paths for erlc and dialyzer should be set on the commandline.

In short:
Do not load '.erlang' file from current directory when starting erlang.
Do not load '.erlang' from tools (escript, erlc, dialyzer) at all.

Document c:erlangrc(PathList) so the user can load '.erlang' manually.

### CODE REVIEW

**Comment by okeuday** [comment]
Getting rid of hidden configuration files makes Erlang systems easier to manage.  Otherwise, it is possible that a server's config gets buried somewhere that lacks visibility.

**Comment by essen** [comment]
Perhaps a good start would be to print a line indicating you are loading one such hidden file.

**Comment by okeuday** [comment]
@essen Avoiding adding noise to the runtime is a good thing.  Adding output about loading a hidden file doesn't make the file not hidden, it simply shows it exists upon every runtime, increasing the amount of information you need to care about.  Putting configuration where configuration should go, shouldn't require knowing about the configuration every time you run something.  That means it is better to get rid of hidden configuration files, than make lots of noise about their use (noise that could be ignored, confusing, and put into output that gets buried in a log file somewhere).

**Comment by essen** [comment]
I'm not so sure. We already ignore some output when starting Erlang most of the time, and yet that output is priceless when trying to debug many issues.

```
Erlang/OTP 20 [erts-9.0.1] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:10] [hipe] [kernel-poll:false]
```

Adding some extra info on this line (if possible) or an extra line before will be as easily ignored when we don't care about it, and immensely useful when we do.

**Comment by okeuday** [comment]
@essen I see the idea of adding output about files that should normally not exist as not useful, just something noisy and definitely not in the same category as all the information currently printed.  Version information, scheduler information, compilation features, etc. is all very very important and succinctly fitting into a single line.  Ensuring the .erlang files aren't used by default is the safest option.

File: erts/doc/src/erl.xml [hunk 1]
```diff
@@ -1633,8 +1633,7 @@
       <tag>The <c>.erlang</c> startup file</tag>
       <item>
         <p>When Erlang/OTP is started, the system searches for a file named
-          <c>.erlang</c> in the directory where Erlang/OTP is started. If not
-          found, the user's home directory is searched for an <c>.erlang</c>
+          <c>.erlang</c> in the user's home directory is searched for an <c>.erlang</c>
           file.</p>
```
Comment (legoscia) [inline erts/doc/src/erl.xml]: The latter half of this sentence can be dropped. It should read:

> When Erlang/OTP is started, the system searches for a file named `.erlang` in the user's home directory.

Comment (dgud) [reply erts/doc/src/erl.xml]: Thanks, fixed.

**Pull request merged**